hyperspy/io_plugins/emd.py

```diff
@@ -0,0 +1,392 @@
+    are saved in four dictionaries (`user`, `microscope`, `sample`, `comments`). To print
+    relevant information about the EMD instance use the :func:`~.print_info` function. EMD
+    instances can be loaded from and saved to emd-files, an hdf5 standard developed at Lawrence
+    Berkeley National Lab (http://emdatasets.lbl.gov/).
+
+    Attributes
+    ----------
+    signals: dictionary
+        Dictionary which contains all datasets as :class:`~hyperspy.signal.Signal` instances.
+    user: dictionary
+        Dictionary which contains user related metadata.
+    microscope: dictionary
+        Dictionary which contains microscope related metadata.
+    sample: dictionary
+        Dictionary which contains sample related metadata.
+    comments: dictionary
+        Dictionary which contains additional commentary metadata.
+
+    """
+
+    _log = logging.getLogger(__name__)
+
+    def __init__(self, signals=None, user=None, microscope=None, sample=None, comments=None):
+        self._log.debug('Calling __init__')
+        # Create dictionaries if not present:
+        if signals is None:
+            signals = {}
+        if user is None:
+            user = {}
+        if microscope is None:
+            microscope = {}
+        if sample is None:
+            sample = {}
+        if comments is None:
+            comments = {}
+        # Make sure some default keys are present in user:
+        for key in ['name', 'institution', 'department', 'email']:
+            if key not in user:
+                user[key] = ''
+        self.user = user
+        # Make sure some default keys are present in microscope:
+        for key in ['name', 'voltage']:
+            if key not in microscope:
+                microscope[key] = ''
+        self.microscope = microscope
+        # Make sure some default keys are present in sample:
+        for key in ['material', 'preparation']:
+            if key not in sample:
+                sample[key] = ''
+        self.sample = sample
+        # Add comments:
+        self.comments = comments
+        # Make sure the signals are added properly to signals:
+        self.signals = {}
+        for name, signal in signals.iteritems():
+            self.add_signal(signal, name)
+
+    def __getitem__(self, key):
+        # This is for accessing the raw data easily. For the signals use emd.signals[key]!
+        return self.signals[key].data
+
+    def _write_signal_to_group(self, signal_group, signal):
+        self._log.debug('Calling _write_signal_to_group')
+        # Save data:
+        dataset = signal_group.create_group(signal.metadata.General.title)
+        maxshape = tuple(None for _ in signal.data.shape)
+        dataset.create_dataset('data', data=signal.data, chunks=True, maxshape=maxshape)
+        # Iterate over all dimensions:
+        for i in range(len(signal.data.shape)):
+            key = 'dim{}'.format(i+1)
+            offset = signal.axes_manager[i].offset

```

Comment:
I changed it into the first format, suggested by @to266 for now, I think this will be correct even if the convention for the public interface changes, right?